frontend/src/components/smart/PvPArena.vue

```diff
@@ -170,6 +180,13 @@ export default {
       this.$emit('leaveMatchMaking');
     },
 
+    async updateSecondsBeforeSeason() {
+      this.currentRankedSeason = await this.contracts().PvpArena.methods.currentRankedSeason().call({ from: this.defaultAccount });
```

Comment:
Don't use direct contract calls in vue files, keep it in store.ts, it applies to every call here

Reply:
@Chriserus I disagree with this, passing all calls through the store has no real value if no app state is affected. The main issue with the previous implementation of this feature was that it's whole state was passed through the store even though it was specific to this view.

Also that meant we had to manually reset the state on route changes.

Reply:
"no real value", it's basic single responsibility principle. It's the same with any api calls, why do we create services for that? Won't it work directly from component? Ofc it will, but will be non-extendable and the SRP will be broken. That's why we have layers (components call api through services, web3 through store.ts) and at least some sort of architecture in our project. I know that single, huge store.ts is not ideal, I'm aware of that, but it's better to continue extracting calls there then to litter components with a responsibility that shouldn't be handled there.

Also, the solution you guys provided is error prone because it's not checking contracts or defaultAccount existence and they're not always available (see checks in every method in store).

Happy to discuss it further.

Reply:
Also, "that meant we had to manually reset the state on route changes", I'm not talking about handling state of PvP in store. I just mean that when you have a simple getter like the line we're commenting under, it should be called this way:

this.currentRankedSeason = await this.currentRankedSeason(); 
In which, currentRankedSeason is an action imported from the store (ideally with an interface here showing what are the arguments and what is the output)

Reply:
Example interface from another class: 
```
interface StoreMappedActions {
  fetchShields(shieldIds: (string | number)[]): Promise<void>;
  fetchWeapons(weaponIds: (string | number)[]): Promise<void>;
  fetchCharacters(characterIds: (string | number)[]): Promise<void>;
}
```

Reply:
Replying to this now, sorry for the delay.

About SRP: I think this is not a case of SRP violation, one store doing every web3 call has multiple responsibilities. I think there's clear value in extracting reusable code, but these calls are literally only used for this component. I do think there's a strong case here in terms of this PR being inconsistent with the rest of the codebase though.

About "Checking for defaultAccount existence (should be)". I guess we assumed this values were guaranteed to be set by the time the route rendered. If not then yup, we have to do that, though I'd say we should consider this in the future as part of an architecture improvement, we shouldn't be repeating this logic on every view if we can extract it.

I think moving the calls themselves to the store is OK, if that's how we are doing it everywhere else, but the previous implementation had horrible issues with the **state** being global. If your request is just about moving the web3 calls into actions then we can definitely do that.

TL;DR;

* Happy to move web3 calls to actions for consistency
* I'm against moving view-specific state to the store (though I think I misunderstood, and you weren't suggesting this)
* I have to dissagree with this being a SRP issue in this feature, and I'd even argue it *is* a big issue on the store itself, as it should be subdivided into modules that do respect that principle (right now being at 4k+ lines), so if anything, that's where we should be looking to make improvements regarding separation of concerns first.

One last thing though. I think we should merge this PR and tackle the migration of web3 calls to actions. Because there's already multiple instances of these merged, and I think it is better if we tackle this on a separate PR

Reply:
I'm aware of our overall architectural issues with having just one store, we're on the same page here. I didn't want you to move any pvp-specific states to the store, that's ok.

Just for the sake of consistency, please move all the web3 calls to store.ts, so we have it all in one place and not scattered in components.

I'm happy with doing it on a different PR in the future, I just want you to make sure it won't get lost due to new features or something, cleanup should take place in not-so-distant future.

As my other comments were formatting-related I will approve the PR and you can merge it in appropriate time.

@seiyria tagging you here, as we had a nice discussion with Rafael and I want it to be visible

Reply:
Thanks for the help @Chriserus , will do that